src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -74,12 +78,59 @@ public IHostBuilder CreateHostBuilder(IConsole console, string[] urls, string[]
                     builder.AddJsonFile(UserSettingsPath, optional: true, reloadOnChange: true);
                     builder.AddJsonFile(SharedSettingsPath, optional: true, reloadOnChange: true);
 
-                    builder.AddKeyPerFile(SharedConfigDirectoryPath, optional: true);
+                    //HACK Workaround for https://github.com/dotnet/runtime/issues/36091
+                    //KeyPerFile provider uses a file system watcher to trigger changes.
+                    //The watcher does not follow symlinks inside the watched directory, such as mounted files
+                    //in Kubernetes.
+                    //We get around this by watching the target folder of the symlink instead.
+                    //See https://github.com/kubernetes/kubernetes/master/pkg/volume/util/atomic_writer.go
+                    string path = SharedConfigDirectoryPath;
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
Could you scope this even further with RuntimeInfo.IsInDockerContainer or add another one for detecting when running in Kubernetes?